hbase_storage/src/main/java/com/splicemachine/si/data/hbase/coprocessor/SIObserver.java

```diff
@@ -92,6 +97,45 @@
     protected boolean authTokenEnabled;
     protected Optional<RegionObserver> optionalRegionObserver = Optional.empty();
 
+    /**
+     * We need to keep track of compacted files in HFiles in order to avoid forgetting about any of them when
+     * recovering from a crash, since the compaction marker might have been lost due to a WAL roll or another crash
+     * See DB-10896 and HBASE-20724 for more details
+     */
+    @Override
+    public void postOpen(ObserverContext<RegionCoprocessorEnvironment> c) {
```

Comment:
Awesome fix!   So glad this issue is finally understood.  Do you think we need to do something similar in paths that don't use snapshot isolation, for example, table and schema backup, so that the backup doesn't include the compacted store files?

Reply:
Good point @msirek , we are already doing it for Full backups (see https://github.com/splicemachine/spliceengine-ee/blob/master/splice_backup/src/main/java/com/splicemachine/backup/HBaseBackupUtils.java#L384 ) but maybe we are missing some cases.
@jyuanca do we need the same thing for Incremental backups and maybe table/schema backups?

Reply:
@dgomezferro Database/table/schema backup excludes all compacted HFiles. If we can make sure compacted HFiles do not come alive after an HBase restart and recover, backup would not copy these files.